wallet/src/de/schildbach/wallet/ui/dashpay/CreateIdentityService.kt

```diff
@@ -317,19 +305,35 @@ class CreateIdentityService : LifecycleService() {
         platformRepo.resetIdentityCreationStateError(blockchainIdentityData)
 
         val wallet = walletApplication.wallet!!
-        val password = securityGuard.retrievePassword()
-
 
-        val encryptionKey = deriveKey(backgroundHandler, wallet, password)
+        val encryptionKey = platformRepo.getWalletEncryptionKey() ?: throw IllegalStateException("cannot obtain wallet encryption key")
```

Comment:
> Could be a good idea. `DecryptSeedTask` seems to be quite simple class whose purpose is to run `decrypt` in the background. We can just launch it in a `withContext` instead. This method that wraps it in a coroutine is also unnecessary:
> 
> ```
> /**
> * Wraps callbacks of DecryptSeedTask as Coroutine
> */
> private suspend fun decryptSeed(handler: Handler, wallet: Wallet, encryptionKey: KeyParameter): DeterministicSeed 
> ```
> 
> Instead, we can just run it like this:
> 
> ```
> withContext(Dispatchers.Default) {
>    val decryptedSeed = seed.decrypt(keyCrypter, null, keyParameter)
> }
> ```
> 
> For `DeriveKeyTask`, there is an analog function in the `SecurityFunctions` that can also be run in a context.

I didn't follow these recommendations yet.  There was already a `PlatformRepo.getWalletEncryptionKey()` that I decided to use in `CreateIdentityService`.  And I added a similar method to get the Seed.

Ideally, though, PlatformRepo should be handling Platform functions and not wallet or security functions.  Since this PR has bene tested already, I propose that this suggestion above be part of some refactoring on the next DashPay / CoinJoin related story.